Fix part of #13822: Added beam jobs for exploration validation

Lawful2002 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes part of #13822.
2. Beam job for Exploration category should be defined in constants.ts.
3. Beam job for Exploration scaled avg rating should be 0 - 5 inclusive.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
<img width="458" alt="image" src="https://user-images.githubusercontent.com/73544247/159072910-4c401477-3396-4301-a1a6-a7c8b9ae1a2b.png">
<img width="501" alt="image" src="https://user-images.githubusercontent.com/73544247/159072955-b641d3f1-c342-4eb3-84e7-7edf1e803538.png">
<img width="959" alt="image" src="https://user-images.githubusercontent.com/73544247/160268716-e2dc7098-10ed-4580-9ec7-fb3f96316e56.png">
 ```Total explorations: 25, Curated Lessons: 4```


<!--
Add videos/screenshots of the user-facing interface in various display sizes (mainly phone, tablet, and desktop display size) to demonstrate that the changes made in this PR work correctly. Please also include videos/screenshots of the developer tools browser console, so that we can be sure that there are no console errors.
If the PR changes the UI in any of the files listed in rtl_css.py (i.e, those that have a separate .rtl.css file for styling), make sure to add screenshots with the site language set to Arabic as well.
If the changes in your PRs are autogenerated via a script and you cannot provide proof for the changes then please leave a comment "No proof of changes needed because {{Reason}}".
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

Lawful2002 committed:
added beam job

Lawful2002 committed:
added test for the beam job

Lawful2002 committed:
fixed linting

Lawful2002 committed:
fixed typos

vojtechjelinek commented:
@lkbhitesh07 Can you please take a first pass? Also, should we place all exploration validation jobs into one file?

lkbhitesh07 commented:
Everything looks good, just took a pass.
Yes @vojtechjelinek we should make a file with all batch jobs related to the Exploration, what we can do is for all the `Jobs` which will be removed after successful checks can be in a separate file so that it will be easy to remove and the rest can be accumulated in one file. Please do let me know what do you think about this?
Thanks

vojtechjelinek commented:
> Everything looks good, just took a pass. Yes @vojtechjelinek we should make a file with all batch jobs related to the Exploration, what we can do is for all the `Jobs` which will be removed after successful checks can be in a separate file so that it will be easy to remove and the rest can be accumulated in one file. Please do let me know what do you think about this? Thanks

I don't want to juggle with multiple files, lets just have one for the validation checks and remove the jobs from it when needed.

lkbhitesh07 commented:
> I don't want to juggle with multiple files, lets just have one for the validation checks and remove the jobs from it when needed.

Make sense, lets's do that in a single file @vojtechjelinek. I can see that @Lawful2002 has raised another PR of beam job related to `Exploration` [here](https://github.com/oppia/oppia/pull/14997) great job btw, what I would suggest is after getting successful reviews on both of these PRs make sure to combine them in one as per the above discussion @Lawful2002.
Do let me know what do you guys think?
Thanks

vojtechjelinek commented:
> > I don't want to juggle with multiple files, lets just have one for the validation checks and remove the jobs from it when needed.
> 
> Make sense, lets's do that in a single file @vojtechjelinek. I can see that @Lawful2002 has raised another PR of beam job related to `Exploration` [here](https://github.com/oppia/oppia/pull/14997) great job btw, what I would suggest is after getting successful reviews on both of these PRs make sure to combine them in one as per the above discussion @Lawful2002. Do let me know what do you guys think? Thanks

Yeah, I think we can combine both these PRs in one PR.

Lawful2002 committed:
merged two beam jobs in one file

Lawful2002 committed:
fixed linting

Lawful2002 commented:
@lkbhitesh07 I have added the beam job (and its test) from PR #14997 into this PR. PTAL. Thanks!

lkbhitesh07 reviewed (changes requested):
Looks good, just a few changes are needed @Lawful2002 - 
1. Rename the file to `exploration_validation_jobs.py` and `exploration_validation_jobs_test.py`
2. Update the screenshots as per new changes and make sure that it has both valid and invalid explorations so that we can see if the error message is showing correctly or not.

Thanks

Lawful2002 committed:
renamed files

Lawful2002 commented:
@lkbhitesh07 Done. I've updated the file names and added screenshots. However for scaled avg rating I'm unable to create an invalid exp. with invalid ratings. PTAL

lkbhitesh07 reviewed (approved):
Looks good to me, Thanks @Lawful2002

Lawful2002 committed:
fixed formatting

Lawful2002 committed:
Merge branch 'develop' of https://github.com/oppia/oppia into beam-jobs

vojtechjelinek reviewed (changes requested):
Thanks! Looks good. Please submit a backup server testing request.

Lawful2002 commented:
Submitted a backup server testing request. Thanks!

vojtechjelinek commented:
@Lawful2002 Hey, I tested the jobs on the backup server and both are working. For `GetExpWithInvalidRatingJob` all explorations are successful, for `GetExpWithInvalidCategoryJob` can you please modify the jobs so that empty category is also accepted, because now we get a bunch of errors where the category is empty.

Lawful2002 committed:
empty categories are now valid

Lawful2002 committed:
Merge branch 'develop' of https://github.com/oppia/oppia into beam-jobs

Lawful2002 committed:
fixed linting

Lawful2002 commented:
@vojtechjelinek I modified the job (and the test) to accept 'empty' category as valid. PTAL.

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,150 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for exploration"""
```
```suggestion
"""Validation jobs for exploration models."""
```

Lawful2002 replied:
Done!

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,150 @@
+            | 'Combine exploration id and category' >> beam.Map(
+                lambda exp: (exp.id, exp.category))
+            | 'Filter exploraton with category not in constants.ts' >>
+                beam.Filter(
+                    lambda exp: len(exp[1]) > 0 and
+                    not exp[1] in constants.ALL_CATEGORIES
+                    )
```
```suggestion
                beam.Filter(lambda exp: (
                    len(exp[1]) > 0 and
                    not exp[1] in constants.ALL_CATEGORIES
                ))
```

Lawful2002 replied:
Done!

vojtechjelinek reviewed (changes requested):
Thanks! Took a pass.

vojtechjelinek commented:
@lkbhitesh07 Do you think that empty category is correct?

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

Lawful2002 commented:
@vojtechjelinek @lkbhitesh07  I think only un-published explorations could have an empty category. It is not possible to publish an exploration without setting its category.
<img width="960" alt="image" src="https://user-images.githubusercontent.com/73544247/156743065-88397eb0-7f5b-4c29-aaef-f6c01eb9b830.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/73544247/156743131-976e1746-ecd6-4036-86dc-5aa34c33dd94.png">

vojtechjelinek commented:
@Lawful2002 Can you implement that in you job?

Lawful2002 commented:
@vojtechjelinek Do you mean, I need to run this job only on explorations that have been published (and remove the `empty` check)?

vojtechjelinek commented:
@Lawful2002 Yep.

Lawful2002 committed:
check only published exps

Lawful2002 committed:
Merge branch 'beam-jobs' of https://github.com/Lawful2002/oppia into beam-jobs

Lawful2002 committed:
check only published exps

Lawful2002 commented:
@vojtechjelinek Done. PTAL

Lawful2002 committed:
removed unnecessary stuff

lkbhitesh07 commented:
> @lkbhitesh07 Do you think that empty category is correct?

I think the empty category is not correct and as @Lawful2002 mentioned that we are not allowed to publish an exploration without a category so that means we are maintaining the category field for future explorations and we might need to decide what to do with the explorations with no category.
So as per the above discussion, we will be checking it for only published explorations and if there are any errors and curated lessons are part of it then we need to handle them separately. I think for the unpublished explorations when in future the creator will try to publish it then it will be handled by frontend validation.
What do you think @vojtechjelinek?

Lawful2002 commented:
@lkbhitesh07 @vojtechjelinek I have added the changes so that only published explorations are checked for their category. PTAL.

lkbhitesh07 commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,154 @@
+            )
+        )
+
+        exp_ids_with_category_not_in_contants = (
```
```suggestion
        exp_ids_with_category_not_in_constants = (
```

Lawful2002 replied:
Done

lkbhitesh07 commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,154 @@
+        )
+
+        report_number_of_invalid_exps = (
+            exp_ids_with_category_not_in_contants
```
```suggestion
            exp_ids_with_category_not_in_constants
```

Lawful2002 replied:
Done

lkbhitesh07 commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,154 @@
+        )
+
+        report_invalid_ids_and_their_category = (
+            exp_ids_with_category_not_in_contants
```
```suggestion
            exp_ids_with_category_not_in_constants
```

Lawful2002 replied:
Done!

lkbhitesh07 reviewed (changes requested):
Just a few nits.
Thanks

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Hitesh Tomar <44300735+lkbhitesh07@users.noreply.github.com>

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Hitesh Tomar <44300735+lkbhitesh07@users.noreply.github.com>

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Hitesh Tomar <44300735+lkbhitesh07@users.noreply.github.com>

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,154 @@
+            exp_ids_with_category_not_in_constants
+            | 'Save info on invalid exps' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                    'The id of exp is %s and its category is %s'
```
```suggestion
                    'The id of exp is "%s" and its category is "%s"'
```

vojtechjelinek replied:
Lets do this so that it is clear when the category is empty.

Lawful2002 replied:
Done!

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,154 @@
+        published_explorations = (
+            total_explorations
+            | 'Get published explorations' >> beam.Filter(
+                lambda exp: exp.first_published_msec is not None
```
This is not a good filter. Please use the `status` field.

Lawful2002 replied:
Done!

vojtechjelinek reviewed (changes requested):
Thanks! Took a pass.

lkbhitesh07 reviewed (approved):
Changes look good to me. Thanks, @Lawful2002

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

Lawful2002 committed:
improved filter condition

vojtechjelinek commented:
@Lawful2002 Is this ready for another run on the backup server?

Lawful2002 commented:
@vojtechjelinek Yup, I'll send in a job request for this and also the skill validation.

vojtechjelinek commented:
Here are the results for this job, @lkbhitesh07 can you handle the next steps? 
```
The id of exp is "2_VzsGOl_H9A" and its category is "training"
The id of exp is "32om7COBJm3V" and its category is "Training"
The id of exp is "YASjVQLtSl-k" and its category is "Religion"
The id of exp is "Ym7iKwBurnQ5" and its category is "Training"
The id of exp is "ZmUhCFe4tErT" and its category is "Religion"
The id of exp is "_mXijSV45q1i" and its category is "training"
The id of exp is "alwazNxQ6ltl" and its category is "Fonction"
The id of exp is "b0VUqVdY5Y3u" and its category is "Training"
The id of exp is "bFJBQs2t0PMN" and its category is "Training"
The id of exp is "bPylJepTMbBi" and its category is "training"
The id of exp is "bXxPOqQzI3N_" and its category is "Training"
The id of exp is "cCGb3LSyN8xG" and its category is "Training"
The id of exp is "d29wfhnznmuD" and its category is "Tutorial"
The id of exp is "dBRiEv5NtgZL" and its category is "Training"
The id of exp is "dOcsg0U9b5uX" and its category is "Training"
The id of exp is "ez8grLSO0mgP" and its category is "Englisch"
The id of exp is "fECG_zfYfZWD" and its category is "training"
The id of exp is "fgEJEuHU8nMh" and its category is "Religion"
The id of exp is "iIz4BZ58YJeW" and its category is "Diabetes"
The id of exp is "ikNOgV0xySJi" and its category is "Training"
The id of exp is "Ig4_L8ziaZpr" and its category is "Training"
The id of exp is "K88NKBvhcV42" and its category is "Pharmacy"
The id of exp is "KAh1weuZ-ckB" and its category is "Training"
The id of exp is "Xz917UfzVVda" and its category is "College Success"
The id of exp is "hggOmiJ7bNGF" and its category is "College Success"
The id of exp is "M4fUuWRy79Zn" and its category is "College Success"
The id of exp is "q9YB_2NG2tMm" and its category is "College Success"
The id of exp is "T0Ug5h9i5A6a" and its category is "computer science"
The id of exp is "MJYbPoHaS0zc" and its category is "online education"
The id of exp is "V-hwjvFJud6S" and its category is "Curriculum design"
The id of exp is "OHd5UZsgMN37" and its category is "Trades Mathematics"
The id of exp is "syCUeQ3w1ypH" and its category is "电力知识普及"
The id of exp is "zZ0pSgqCMb_O" and its category is "English Composition"
The id of exp is "p3VjWaU7t66G" and its category is "student rep training"
The id of exp is "p8RkxcETAeYi" and its category is "Student Rep Training"
The id of exp is "0mgpwPIJ6vyg" and its category is "Tanulasi Lehetosegek"
The id of exp is "mnXq-AhhD4xH" and its category is "UMSU Society Training"
The id of exp is "zAo2xHJMHIRI" and its category is "Academic Representation"
The id of exp is "qFocZicvbdBS" and its category is "Jecjedjejcjexjjexjexjjecjecnjecnejxe"
The id of exp is "9waI7o16SDBc" and its category is "YUSU Student Group Signatory Training"
The id of exp is "Ul1DnWQ5EUb4" and its category is "YUSU Student Group Signatory Training"
The id of exp is "EkbYpzlYLmbC" and its category is "YUSU Student Group Signatory Training"
The id of exp is "MTWCYAlrDRZv" and its category is "Training"
The id of exp is "puWdLCYQTtJC" and its category is "Training"
The id of exp is "qNir9tBBqdEU" and its category is "Religion"
The id of exp is "qPOAnG3iiXXI" and its category is "Schulung"
The id of exp is "tFp1kzGfKXt6" and its category is "Training"
The id of exp is "5URDuUhQ-gja" and its category is "Afrikaans"
The id of exp is "AGjyHA8upGbA" and its category is "webdesign"
The id of exp is "PZYduhJbuLtB" and its category is "Elearning"
The id of exp is "X58yQCo0CofZ" and its category is "Cosmology"
The id of exp is "EVc6bZ9aswbp" and its category is "education"
The id of exp is "HBKDiHkxVOos" and its category is "Campaigns"
The id of exp is "Xsn7O5aYuHdv" and its category is "Gardening"
The id of exp is "axqUeKJEF8Yw" and its category is "webdesign"
The id of exp is "brIUr4aMwNjH" and its category is "Literaura"
The id of exp is "fGBfw2h8aZRl" and its category is "grammaire"
The id of exp is "hxcuPFC7G3CD" and its category is "Elections"
The id of exp is "IB9-grMUKtwT" and its category is "Catechese"
The id of exp is "9bNLEdFEQ_TC" and its category is "bioquimica"
The id of exp is "PMmw85-cFLWv" and its category is "Catecheses"
The id of exp is "kKQsKTocYiQX" and its category is "Psychology"
The id of exp is "uxsjP-5MQAJD" and its category is "Management"
The id of exp is "vFr563Qwx8Hz" and its category is "Decoration"
The id of exp is "zq2gRy4oKt8I" and its category is "Healthcare"
The id of exp is "Fh1lvvLFcLZY" and its category is "bioquimica"
The id of exp is "hhsOPZgdrzuW" and its category is "Psychology"
The id of exp is "5gkBmY0W4XEc" and its category is "social work"
The id of exp is "6dz_rWig579S" and its category is "Life Skills"
The id of exp is "6yTt3c2NVnFv" and its category is "Life Skills"
The id of exp is "9eh7z_FTzsLA" and its category is "Life Skills"
The id of exp is "Oue1mQVhHKYb" and its category is "Linguistics"
The id of exp is "PnlVuIimgygz" and its category is "Life Skills"
The id of exp is "QPfAH0GJUQAx" and its category is "Photography"
The id of exp is "TxKq-_deand1" and its category is "Life Skills"
The id of exp is "k68tKfzqr_kD" and its category is "Life Skills"
The id of exp is "v_ye4IZtvT1L" and its category is "Linguistics"
The id of exp is "VOqb6dCyB2-j" and its category is "Life Skills"
The id of exp is "VkQ8BpmZwzir" and its category is "Life Skills"
The id of exp is "0Kwt-uCpdXaY" and its category is "Information"
The id of exp is "YShSAJUia0V2" and its category is "Operational"
The id of exp is "atuSL6ejtWz8" and its category is "Ifops Tasks"
The id of exp is "bE1aDYFlsckB" and its category is "Linguistics"
The id of exp is "gJ2IS_-Xt87q" and its category is "Linguistics"
The id of exp is "hGaXZSVXthaG" and its category is "Exploration"
The id of exp is "IUhjjNCpnIAf" and its category is "Life Skills"
The id of exp is "JUtTbfH41QI-" and its category is "Life Skills"
The id of exp is "sYPKoexVx0tN" and its category is "YUSU Student Group Signatory Training"
The id of exp is "smrf8Fe7dYwd" and its category is "YUSU Student Group Signatory Training"
The id of exp is "b-cTR-8optWO" and its category is "Cooking"
The id of exp is "bTz2r6d0n4le" and its category is "Cooking"
The id of exp is "ciXdXjwM4Ltx" and its category is "Society"
The id of exp is "eU0wkUXE3s2H" and its category is "science"
The id of exp is "eYDhAgrNeu5H" and its category is "Kitchen"
The id of exp is "i5RfVfgOeVrt" and its category is "website"
The id of exp is "KBCNwchWmIYb" and its category is "english"
The id of exp is "qi5slfMid8Hx" and its category is "Cooking"
The id of exp is "r8TlylkqCNIP" and its category is "Ciencia"
The id of exp is "6qOGXvJewUYd" and its category is "Training"
The id of exp is "7nt7kQwfzdl_" and its category is "training"
The id of exp is "7vlL4xGBYYov" and its category is "Training"
The id of exp is "7xJJtlo01McC" and its category is "training"
The id of exp is "7zTDRnZgSu7b" and its category is "Training"
The id of exp is "8CzqBfie_eg5" and its category is "Religion"
The id of exp is "8aMQpLnUdFmO" and its category is "Religion"
The id of exp is "9vo-s9Ha9opK" and its category is "Training"
The id of exp is "AJwWj8FLhRcW" and its category is "Training"
The id of exp is "OFtv7mw2lmk2" and its category is "Religion"
The id of exp is "Or6vsj8vOEyL" and its category is "Training"
The id of exp is "OrQdbrX50s4t" and its category is "Training"
The id of exp is "RtrqJfWtJspB" and its category is "Religion"
The id of exp is "SCy2B7Tjtb2F" and its category is "Training"
The id of exp is "psq427fTb5pK" and its category is "Life Skills"
The id of exp is "wmdkWgLQkBFH" and its category is "biochemistry"
The id of exp is "C4DYINTujZmw" and its category is "Safeguarding"
The id of exp is "HGMBneKELgKY" and its category is "Safeguarding"
The id of exp is "23uInRQE-Tn8" and its category is "Safeguarding"
The id of exp is "Z8rNfAMImrKm" and its category is "Safeguarding"
The id of exp is "rQYXru4iyREY" and its category is "Social Media"
The id of exp is "ly7o4bMGpvye" and its category is "Modern Issues"
The id of exp is "gflNkTwc33M7" and its category is "cooking class"
The id of exp is "kHOIgqaUPRSS" and its category is "Skilled Trades"
The id of exp is "xTc_GJ0m2AQI" and its category is "Skilled Trades"
The id of exp is "z8kLoEpWTZ9E" and its category is "Skilled Trades"
The id of exp is "zGVE-B3j7-JO" and its category is "Skilled Trades"
The id of exp is "CjVHAJSNz1P_" and its category is "Computer Virus"
The id of exp is "2Xsx4v7T3Puv" and its category is "Skilled Trades"
The id of exp is "jYLAX5_LuMlY" and its category is "Skilled Trades"
The id of exp is "s9pvFc008PVV" and its category is "Skilled Trades"
The id of exp is "w2_pbtWiZBt1" and its category is "College Success"
The id of exp is "yx_jWI0vz_Ss" and its category is "College Success"
The id of exp is "FaF6vuK_F8SF" and its category is "College Success"
The id of exp is "4fWQxdOztPsf" and its category is "College Success"
The id of exp is "SMfVLsh65rpp" and its category is "Training"
The id of exp is "Txv_CRKQTgdX" and its category is "training"
The id of exp is "nmTQJfXGOB1C" and its category is "Training"
The id of exp is "nyHHqX3u3mbv" and its category is "training"
The id of exp is "otW9TNz9D_ov" and its category is "Religion"
The id of exp is "oulVqa4IPOfn" and its category is "Training"
The id of exp is "ovPLruY-L6lB" and its category is "Training"
The id of exp is "uSgtojF9RFwz" and its category is "Training"
The id of exp is "ugWsoPMYBFUy" and its category is "Training"
The id of exp is "ujd8BeDhCTqM" and its category is "Bicycles"
The id of exp is "x1-XXNas1Jv9" and its category is "Training"
The id of exp is "xCyqICtjDxMA" and its category is "Finances"
The id of exp is "U5MouYkEwabr" and its category is "training"
The id of exp is "UCRts3sU0v_t" and its category is "Training"
The id of exp is "WTtPKNONzuqC" and its category is "fraction"
The id of exp is "Byb0EnDskWbD" and its category is "Diabetes"
The id of exp is "EQdY5sskrAcG" and its category is "Internet"
The id of exp is "FYxuBgTOaLnr" and its category is "Diabetes"
The id of exp is "HERGlYeCYE4F" and its category is "Geologia"
The id of exp is "HICNva73tabL" and its category is "Finances"
The id of exp is "HOy8LkguQca9" and its category is "Training"
The id of exp is "28VIEB5VzXaN" and its category is "Training"
The id of exp is "2P1KFA7mGiQd" and its category is "Religion"
The id of exp is "Nbdq0RxtpumI" and its category is "Home"
The id of exp is "q4xyVOWUUQO4" and its category is "Quiz"
The id of exp is "5vGHvVJbuXqL" and its category is "Other"
The id of exp is "8OU31jU1iQh9" and its category is "Other"
The id of exp is "Ts0VswFck3VE" and its category is "Media"
The id of exp is "TxqDm6UyVTQ9" and its category is "musik"
The id of exp is "mcJkuWKwabBQ" and its category is "Trial"
The id of exp is "nVB1D39-OIT-" and its category is "whmis"
The id of exp is "wneXeF1KovhW" and its category is "Other"
The id of exp is "VbQmdtrVnO4o" and its category is "Other"
The id of exp is "EsJFa7PFmI4e" and its category is "Other"
The id of exp is "Et4qwuG0h5ab" and its category is "other"
The id of exp is "-_dOKZ2T8M5Z" and its category is "Ocean"
The id of exp is "0eXE2-kW6x1s" and its category is "Other"
The id of exp is "4_IVMXbIkyNs" and its category is "Other"
The id of exp is "c-BizEiHj4S3" and its category is "mathe"
The id of exp is "gJQQSjTwzNAp" and its category is "Other"
The id of exp is "glYBK8Ir6vRe" and its category is "Other"
The id of exp is "iBYF1g27OAEg" and its category is "Media"
The id of exp is "isTl3eaG91AV" and its category is "Other"
The id of exp is "striajBnOYK6" and its category is "Apnea"
The id of exp is "Qzd3pDSFpOiY" and its category is "social"
The id of exp is "SEcUHxug-GAc" and its category is "basics"
The id of exp is "Uj1A5TdaBo0f" and its category is "IPower"
The id of exp is "4uivHFExXjAz" and its category is "HR"
The id of exp is "5lsSjXsajTNa" and its category is "Fen"
The id of exp is "RfpR70QFkvSC" and its category is "new"
The id of exp is "nbt_g6gt_01i" and its category is "law"
The id of exp is "vYDiqtV328pk" and its category is "New"
The id of exp is "4NC3ykjjW5-b" and its category is "New"
The id of exp is "52tkAhAbOtl1" and its category is "GPS"
The id of exp is "XPunlwSSXs71" and its category is "fun"
The id of exp is "bmJSYvR_slYS" and its category is "UVV"
The id of exp is "cNfLYValOZbS" and its category is "New"
The id of exp is "dK5MHFjtgE8T" and its category is "new"
The id of exp is "sTzojwfMdXZh" and its category is "Eat"
The id of exp is "smN3jNL75jSn" and its category is "New"
The id of exp is "91yImzbxTg6y" and its category is "Neos"
The id of exp is "Psx9OL3KQVoi" and its category is "test"
The id of exp is "m2y0q8_0mfIC" and its category is "test"
The id of exp is "mqIJoaU9orqB" and its category is "Food"
The id of exp is "nLbkczTdjMT7" and its category is "math"
The id of exp is "nRpqnv1ZmmUN" and its category is "work"
The id of exp is "wJdtUSHyoJKZ" and its category is "Test"
The id of exp is "CxiQmFK4wgSs" and its category is "test"
The id of exp is "EwL3TfvljD4-" and its category is "Home"
The id of exp is "j6613_2IPDTP" and its category is "Food"
The id of exp is "IPOzIelGo3oA" and its category is "Neos"
The id of exp is "BxJ8Z48FQkKq" and its category is "basics"
The id of exp is "C0Kl4skpqxn3" and its category is "French"
The id of exp is "a1tEA2hvty86" and its category is "Gender"
The id of exp is "hQ5Aj1enhEWh" and its category is "Events"
The id of exp is "IluXmxlnucQ9" and its category is "kultur"
The id of exp is "Lgq1s4rDoTm3" and its category is "health"
The id of exp is "LlLwVNCVOYrK" and its category is "logica"
The id of exp is "snhDjZLuG0GW" and its category is "french"
The id of exp is "5uRUmr7eY5Ci" and its category is "Cooking"
The id of exp is "6XOmTOCYd4u3" and its category is "Hobbies"
The id of exp is "9UqJzz3srhY0" and its category is "Zdravie"
The id of exp is "RvAq08Yu2DaF" and its category is "Anglais"
The id of exp is "mJAlDAIgkRvy" and its category is "nouveau"
The id of exp is "mS9oIvC3mgOE" and its category is "anglais"
The id of exp is "tUFEaKJvDQPU" and its category is "Science"
The id of exp is "wiqw8O3iiVi8" and its category is "science"
The id of exp is "wlRWUyAy2vm1" and its category is "Anatomy"
The id of exp is "VKc0YRV2sOaD" and its category is "Hobbies"
The id of exp is "CqKVz5c4qvDn" and its category is "Cooking"
The id of exp is "EKuUL8mSe5xb" and its category is "Jupiter"
The id of exp is "1pBNcnVIMMmx" and its category is "Writing"
The id of exp is "Y03gp1iWI4v9" and its category is "Writing"
The id of exp is "ZMEZ4FwWYeyB" and its category is "Finance"
The id of exp is "aKiwo9SJcVk0" and its category is "Science"
```

Lawful2002 committed:
run ecp category job only on curated lesson

Lawful2002 committed:
Merge branch 'develop' of https://github.com/oppia/oppia into beam-jobs

Lawful2002 committed:
fixed linting

Lawful2002 commented:
@lkbhitesh07 I have updated the category job to only test for curated lessons(explorations). Could you PTAL. Thanks!

lkbhitesh07 commented:
Hey @vojtechjelinek,
Will you please take the first pass on new changes?
Thanks

vojtechjelinek commented:
@lkbhitesh07 Sorry, can you try to take the first pass? I will be able to look at this PR tomorrow.

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,115 @@
+            )
+        )
+
+        curated_lessons = (
```
```suggestion
        curated_explorations = (
```

vojtechjelinek commented on core/jobs/batch_jobs/exploration_validation_jobs.py:
```diff
@@ -0,0 +1,115 @@
+            PCollection. Returns PCollection of invalid explorations with
+            their id and category.
+        """
+        all_curated_lessons = (
```
```suggestion
        all_curated_exploration_ids = (
```

Lawful2002 replied:
Done! (updated the screenshots)

vojtechjelinek reviewed (changes requested):
Looks fine. Please add more proof of changes with the new changes, after that I can run this on the backup server.

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

lkbhitesh07 reviewed (approved):
Thanks, Looks Good

SAEb-ai commented:
Please assign me once you get LGTM from the other reviewers because I need to see the runtime of the shards which might change if you push other requested changes.

Lawful2002 committed:
Update core/jobs/batch_jobs/exploration_validation_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

Lawful2002 committed:
Merge branch 'develop' of https://github.com/oppia/oppia into beam-jobs

Lawful2002 committed:
addressed comments

vojtechjelinek commented:
@Lawful2002 Hey after the update the output of this job is "EXPS SUCCESS: 61".

Lawful2002 commented:
Thanks! @vojtechjelinek. Do we need to keep this job around? (invalid curated exps my be added in the future). Or would it be better to find a way to prevent this from happening in the first place?

vojtechjelinek commented:
> Thanks! @vojtechjelinek. Do we need to keep this job around? (invalid curated exps my be added in the future). Or would it be better to find a way to prevent this from happening in the first place?

Job is not needed. Just make sure that we have all the validations in place for this.

Lawful2002 commented:
Validation added in PR [#15342 ](https://github.com/oppia/oppia/pull/15342)
